src/core/coords.js

```diff
@@ -0,0 +1,13 @@
+export default class Coords {
```

Comment:
Do you actually need that class ?

Reply:
A simple objet {x: 0, y: 0} will suffice and it's a lot easier to test!